builtin/providers/rancher/provider.go

```diff
@@ -18,14 +17,14 @@ func Provider() terraform.ResourceProvider {
 			},
 			"access_key": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("RANCHER_ACCESS_KEY", nil),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("RANCHER_ACCESS_KEY", ""),
 				Description: descriptions["access_key"],
 			},
 			"secret_key": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("RANCHER_SECRET_KEY", nil),
+				Optional:    true,
```

Comment:
The point here is that Rancher doesn't require API keys. You could have a rancher server that does not have access control enabled, so it wouldn't make sense to force the user to provide these values then.